### ISSUE TITLE
关于series_loss ，prior_loss和thresh，逻辑上有些疑问

### ISSUE THREAD

**Issue by fffii** [DAMO-DI-ML/KDD2023-DCdetector#31]
在test的部分，包括(1) stastic on the train set，和(2) find the threshold，我不知道自己理解的对不对，就是在train_loader和thre_loader所对应的数据集上统计loss经过softmax变换之后的大小，经过计算百分位数确定thresh的数值。
然后论文中也提到了 normal点之间在不同view下距离是小的，而anomali之间的距离是大的
<img width="670" alt="截屏2024-01-05 17 05 45" src="https://github.com/DAMO-DI-ML/KDD2023-DCdetector/assets/115461655/24f4da22-93f2-4f7e-82bb-fb8a6fb493cb">
算距离是用了kl散度的公式，那我想问的是，对于
metric = torch.softmax((-series_loss - prior_loss), dim=-1)
两个loss值都比较小的点，是不是就是这些点更加相似，然后相加取负数对应softmax它们的值就越大，
但
pred = (test_energy > thresh).astype(int)
看起来说的是比较大的值超过了thresh，是异常的，这感觉逻辑上和前面说不通啊
非常期待和感谢您的回复

**Comment by yyysjz1997** [comment]
> 在test的部分，包括(1) stastic on the train set，和(2) find the threshold，我不知道自己理解的对不对，就是在train_loader和thre_loader所对应的数据集上统计loss经过softmax变换之后的大小，经过计算百分位数确定thresh的数值。 然后论文中也提到了 normal点之间在不同view下距离是小的，而anomali之间的距离是大的 <img alt="截屏2024-01-05 17 05 45" width="670" src="https://private-user-images.githubusercontent.com/115461655/294449241-24f4da22-93f2-4f7e-82bb-fb8a6fb493cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2NDY1MzIsIm5iZiI6MTcxNTY0NjIzMiwicGF0aCI6Ii8xMTU0NjE2NTUvMjk0NDQ5MjQxLTI0ZjRkYTIyLTkzZjItNGY3ZS04MmJiLWZiOGE2ZmI0OTNjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNFQwMDIzNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDQ5ODY2Y2U5YzdmYjVlMmFiZjM4YjZhYWZjYzEyYzQ2MGRhNWE4NTNjOWZhNGJmNmZlNzk0MmQwNjEwMmJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.03IfZa2FhJtBGjf9KycRh1mviHKodcjKBoidyzi9pdk"> 算距离是用了kl散度的公式，那我想问的是，对于 metric = torch.softmax((-series_loss - prior_loss), dim=-1) 两个loss值都比较小的点，是不是就是这些点更加相似，然后相加取负数对应softmax它们的值就越大， 但 pred = (test_energy > thresh).astype(int) 看起来说的是比较大的值超过了thresh，是异常的，这感觉逻辑上和前面说不通啊 非常期待和感谢您的回复

感谢您的issue！您可能存在误解，这个得分（test_energy）为度量是否异常的指标，这个值越大表示越可能异常，其与两点表征相似与否不同。
